cmd/example-app/main.go

```diff
@@ -278,12 +278,21 @@ func (a *app) handleCallback(w http.ResponseWriter, r *http.Request) {
 		}
 		token, err = oauth2Config.Exchange(ctx, code)
 	case "POST":
-		// Form request from frontend to refresh a token.
+		// Form request from frontend to refresh a token; or login again with hint
 		refresh := r.FormValue("refresh_token")
-		if refresh == "" {
-			http.Error(w, fmt.Sprintf("no refresh_token in request: %q", r.Form), http.StatusBadRequest)
+		idTokenHint := r.FormValue("id_token_hint")
+		if refresh == "" && idTokenHint == "" {
+			http.Error(w, fmt.Sprintf("no refresh_token or id_token_hint in request: %q", r.Form), http.StatusBadRequest)
 			return
 		}
+		if idTokenHint != "" {
+			// redirect to auth URL with the hint, using default scopes
+			scopes := []string{"openid", "profile", "email"}
```

Comment:
Seems odd that these aren't taken from the first page.

Reply:
Yeah I was a bit lazy here. I'm going to try to fix this.

Reply:
It's actually not that simple -- we don't know much of the granted scopes when retrieving the id_token 🤔 Since this is the example app, I haven't spent too much time on implementing this.

Reply:
np. it's not critical.